src/providers/openstack/main.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+"""
+Set the OpenStack provider for the autoscaling module.
+
+Classes:
+    ProviderService
+"""
+
+from typing import Dict
+import openstack
+from providers.openstack.settings import (
+    OPENSTACK_FLAVOR,
+    OPENSTACK_IMAGE,
+    OPENSTACK_KEYPAIR,
+    OPENSTACK_METADATA_KEY,
+    OPENSTACK_METADATA_VALUE,
+    OPENSTACK_NETWORK,
+)
+from providers.schema import BaseProviderService
+
+
+class ProviderService(BaseProviderService):
+    """Service of the Openstack provider."""
+
+    def __init__(self):
+
+        self.connector = openstack.connect(cloud="envvars")
+
+    def list(self) -> Dict[str, str]:
+
+        servers = {}
+
+        for server_object in self.connector.compute.servers():
+            server = server_object.to_dict()
+
+            if (
+                OPENSTACK_METADATA_KEY in server["metadata"] and
+                server["metadata"][OPENSTACK_METADATA_KEY] == OPENSTACK_METADATA_VALUE
+            ):
+                servers[server["id"]] = server["addresses"][OPENSTACK_NETWORK][0]["addr"]
```

Comment:
This line may depend on the order of the IPv6 and IPv4 addresses: to be safe, wouldn't it be better to match a specific one ? Like always take `"version": 4` for example